src/amo/components/AddonDetail.js

```diff
@@ -197,7 +209,10 @@ export class AddonDetailBase extends React.Component {
             {i18n.gettext('Extension Metadata')}
           </h2>
           <AddonMeta addon={addon} />
-          <InstallButton {...this.props} />
+          <InstallButton {...this.props} disabled={!compatible} />
+          {!compatible ? (
+            <AddonCompatibilityError reason={reason} />
```

Comment:
Thanks! I added errors if they weren't supplied because obviously that's needed 😄